File: WowsKarma.Common/Models/DTOs/Replays/ReplayDTO.cs
﻿namespace WowsKarma.Common.Models.DTOs.Replays;

public sealed record ReplayDTO
{
	public Guid Id { get; init; }

	public Guid PostId { get; init; }

	public string DownloadUri { get; init; } = "";
	
	public string? MinimapUri { get; init; }

	public IEnumerable<ReplayPlayerDTO> Players { get; set; } = [];

	public IEnumerable<ReplayChatMessageDTO> ChatMessages { get; set; } = [];
}

File: wowskarma.app/src/app/services/api/models/replay-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ReplayChatMessageDto } from './replay-chat-message-dto';
import { ReplayPlayerDto } from './replay-player-dto';
export interface ReplayDto {
  chatMessages?: null | Array<ReplayChatMessageDto>;
  downloadUri?: null | string;
  minimapUri?: null | string;
  id?: string;
  players?: null | Array<ReplayPlayerDto>;
  postId?: string;
}
